Project.ipynb

```diff
@@ -20,6 +20,841 @@
999/v6pjXrx4Ef/8z/+MP/iDP1B7+oroC+FrR61WQzgcxtHREc6cOYMHHngAL730kqqFHlA/qqv5fZovCJZlEY/HEQgEMDIyIjpAWovzkDv2TOj1+XwekUgEDMO0TIZph5YRn5Tn9yNS/0ad2iySySQikUhLmwWJKMWExYz47mGkyS0WiwWrq6uYmZnBE088ga9//esAwFUNq+XcuXOq30MNfSt8tVoNkUgEsVgMCwsLDQUVehem8CERGz9NSAZI+3w+XLlypW1kqsV5qBFPsqdw9+5dnD17VnZbgtriFj5S9pL6LdVJUPo74bdZTE9PA2gtvU8kEiiVSqJtFkaI+ADofg5GEj5CNpttKJLqlKE5LvSF8PEvan7J/Pz8fIPgEXq9P9cOci5OpxPpdBp+vx9OpxP33XcfBgYGenIOSlKd/F48t9uNixcvtgz5lUIv0496pzr1QuviErltFlarFRaLBUdHR11zszguGO1zq2llePDBBxGNRlsef+yxx/Dwww+rPTVV9IXwAfcEb2dnBwcHB5ifn8fm5qbo3ZVWLuxaYLPZkM1mcfv2bVitVllpQq2Qk+oU6sV7+eWXVVVmapEyJrRb5PtZ+HqBWJvFwcEB0uk0CoVCV90sTOTTHPHJ4emnn9b4bLSjL4Tv6OgIt2/f5vzaOl1AWkV8avvfSFNqoVDA+fPndeulkSI+3erFU+vuQLBYLB2FrV+FD9Av0rBYLFyz9eLiIve42FBnJX1oUjDC926Ec2gmm82euB4+oE+Eb2RkRJLgEbQ2o5Vb5VYsFuH3+1EulzE8PIz5+XnVPz416ax2laH8qtLJyUnRXjw1+3RqI756vY5wOIyDgwPYbDZun4lUIfJ7lIy4+HQbvfvohG4OxYY6k0Ka5jYLfs9ht90suoFRf3fdmtP5zW9+E+9973sRj8fxq7/6q7hy5Qr+7d/+TfPjiNEXwudyuWRFcHa7netfUYNc4SuXywgEAsjn81hZWcH4+DgCgYBmo8/U9uHxae7FE6sqFXu9mmNLhWEYVKtV3Lp1C3Nzc7h582bD4tk8zsvlcqFSqaBUKqm2lDpO6C18Uo9PimOE+tDy+bygmwVfEMWuQ70/PzkHI4p1tyK+t771rXjrW9+q+ftKpS+Er1eefErfhz9AemVlpcHpwQh9eM2vV9KL16uWBOC1Vg+/3w+GYbgotFqtgmVZbsTT6dOnueeXy2Ukk0mk02lsb2+jXC43RBKDg4PH0vngOKC2lcDhcAhOKSkWi8jn8x3bLIwgfEas6ATuFbcsLS3pfRqa0xfCJxeHw6GJ8HXy0+P3EC4tLQmOReulC7sYpLiFoihsb2/L7sXrpfARs1qv14tr167hJz/5ScdIl4zzGh8fRyqVwqVLlwCgoSQ/Eok0OB/wIwmtxkzphd4LfzeiHSluFqTNArg3k3d3d1c3/7t6vW5Y4TP3+I4pSiI+rfb4hASHX/nY3EMo9B5qff3UNqDX63UcHh4iGo1ifX1d1148MQqFAra3t0HTtCyz2uZj8UVWaN4hKcnP5/OIxWIIBAItRRfEJFbvKEIqegsfwzA9uXkQa7MoFou4ffs2bDabJDeLbmDkiO+kOTMAfSJ8culWqpOmaezu7mJvbw9zc3NtWyoIekZ8JAV7dHSE4eFhXLp0SdGFr3VLAp9qtYpAIIBsNou1tTVVTvJSPptYST5/tuXe3h6q1SqXViPRodfrNawY6h3x6e2M4XQ6MTMz03BOYm4W3Wiz0HJAtZaYVZ19hNbCxzAM9vf3sbOz0zBAWgp6CF9zL97Y2BgymYyq6R5aCx9N0wiHw4hGo1heXpZsF9UJJdV1YrMtK5UKN9syHo+jWCxyNjJEEAcGBnS/09e7olDvcWFCczrbuVnw2yyIm4XaNgsjR3xKTKCNTl8In9wFUaveMZvNhkQigZ2dHUxMTODmzZuy9w56KXxivXipVEqTPUItYFkW+/v7iEQiLf2Cneh1H59QWq1er3Op0v39fRQKBe6YHo8HLpdL0aBnNegdcel9fDmWRN1qszCq8HWrnUFv+kL4gN42J7Msi6OjIwQCATgcDly/fl201L8TvXBy5/fiTUxMtPTiqY3YrFar6j1TfvvE2NiYrJsIOeXy3f6N2O12DA8PN6SPGIZBMBhEvV5vqEB0uVwN0WG3WiyMIDx6Rnxqj69Fm4VRha9YLPZsNGIv6Rvh6wVkcQ4EAhgcHMTa2hqy2axi0QO6284gtRdPK1sipdA0jeeffx4ul6vjUO5miJgZRfiEsFqtcLlcGBgYaGixqFQq3MIZjUa5FotueOLpid7C2y1LIjltFizLwuVywel0dnSz6BUsyxq2v1AtfSN8Sixq5NwJplIp+P1+uN1uXLp0CV6vF9lsFslkUukpA9BG+IRSjXJ68bSwJVIifMVikeupu3z5sqKZgVqlrXuNxWKB2+2G2+1u2GMinnjNUQS/xWJwcFBWoYTewnPcIz45iLVZhEIhMAyDcrnc0c2il5BRfyeNvhE+uUidupLNZrG9vQ2bzYbz5883/KA79fHJOQ812Gw2LtWopBdPi1SnnNfznR3W1tZQKBQUD8qVc8Oj56xOqYuLkCceTdNcFBGPxxEMBkHTNDweT0N0KFZwobfw6X18OXt83YCIy/DwcMNNjpibRa/aLIzgk9gtTOEToZPw5fN5bjLI2tqaYMmvWB+fHLSoiLTZbMjn83jllVdQKBRk9+Jp0QAv5TPQNM05aPAb+re2thQvjnJmfR7XIdU2m42bRkMgLRbNBRcOh6MhMpSTNu4W/RTxiSG0xyfWOtOrNot8Pt9zJ5he0TfCJ3fRFLMm4g+QXltbaysgRjC0rVarODw8RCaTwYULFzAxMaGoyrWbER+xpgmHw5iZmWnpbyTpSqXCJyfiOynwWywmJye5x6vVKrdvmEgkUCwWuR4ysr/UawsgvSO+bu3xyT0HKX/zXrZZnNQePqCPhE8uzaJVLpfh9/tBURRWV1dx6tSpjj8gLfbn
```

Comment:
These comments are not very readable because they are too much on the right, write them on the same level as the code